schema.go

```diff
@@ -87,10 +89,16 @@ func (rs *RootSchema) UnmarshalJSON(data []byte) error {
 		if sch, ok := elem.(*Schema); ok {
 			if sch.ID != "" {
 				ids[sch.ID] = sch
-
-				// For the record, I think this is rediculous.
+				// For the record, I think this is ridiculous.
 				if u, err := url.Parse(sch.ID); err == nil {
-					ids[u.Path[1:]] = sch
+					// This is if the identifier is defined as a reference (with #)
+					// i.e. #/properties/firstName
+					// in this case, u.Fragment will have /properties/firstName
+					if strings.HasPrefix(sch.ID, "#") {
```

Comment:
It seems part of the problem derives from this codepath not guarding against a string length of `0`. I'm going to file an issue around that concern, but that fix wouldn't solve _this_ problem of fragments having a `#` prefix that is tripping things up, so I think this should go in.

Reply:
@b5 You are correct, it came to my mind, but the line 90 `if sch.ID != "" {` already does the check. May be I should have added checks to make sure that `u.Fragment` and `u.Path` are non empty before slicing them. Something like below would have solved both the issues

```go
alias := ""
if strings.HasPrefix(sch.ID, "#") {
	alias = u.Fragment
} else {
	alias = u.Path
}

if alias != "" {
	ids[alias[1:]] = sch
}
```